src/coreclr/jit/lir.cpp

```diff
@@ -307,6 +307,26 @@ LIR::ReadOnlyRange::ReadOnlyRange(GenTree* firstNode, GenTree* lastNode) : m_fir
     assert((m_firstNode == m_lastNode) || (Contains(m_lastNode)));
 }
 
+//------------------------------------------------------------------------
+// LIR::ReadOnlyRange::operator=:
+//    Move assignment operator= for LIR ranges.
+//
+// Arguments:
+//    other - The range to move from.
+//
+LIR::ReadOnlyRange& LIR::ReadOnlyRange::operator=(ReadOnlyRange&& other)
```

Comment:
We already have the move constructor on `LIR::ReadOnlyRange` and I needed this version to write some logic cleanly.